common/guestchain/src/manager.rs

```diff
@@ -168,6 +181,43 @@ impl<PK: crate::PubKey> ChainManager<PK> {
         self.pending_block.as_ref()
     }
 
+    pub fn update_config(
+        &mut self,
+        config_payload: UpdateChainConfigPayload,
+    ) -> Result<(), UpdateConfigError> {
+        if let Some(min_validators) = config_payload.min_validators {
+            if min_validators > self.config.min_validators {
+                return Err(UpdateConfigError::MinValidatorsHigherThanExisting);
+            }
+            self.config.min_validators = min_validators;
+        }
+        if let Some(max_validators) = config_payload.max_validators {
+            if max_validators < self.config.max_validators {
+                return Err(UpdateConfigError::MaxValidatorsLowerThanExisting);
+            }
+            self.config.max_validators = max_validators;
```

Comment:
By the way, there’s also self.candidates.max_validators that needs to be updated.

Reply:
got it